plugins/nodes/src/js/data/NodesNetworkClient.ts

```diff
@@ -0,0 +1,35 @@
+import { Observable, merge } from "rxjs";
+import { request, RequestResponse } from "@dcos/http-service";
+import { partition, tap } from "rxjs/operators";
+
+export interface NodeNetwork {
+  updated: string;
+  public_ips: string[];
+  private_ip: string;
+  hostname: string;
+}
+
+export type NodesNetwork = NodeNetwork[];
+
+export function fetchNodesNetwork(): Observable<RequestResponse<NodesNetwork>> {
+  const [success, error] = partition(
+    (response: RequestResponse<NodesNetwork>) => response.code < 300
+  )(request("/net/v1/nodes"));
+
+  return merge(
+    success,
+    error.pipe(
+      tap((response: RequestResponse<NodesNetwork>) => {
+        const responseMessage =
+          response.response && typeof response.response === "object"
+            ? JSON.stringify(response.response)
+            : response.response;
+        throw new Error(
+          `Network Nodes API request failed: ${response.code} ${
```

Comment:
Shall we wrap it with `i18n`? Would that approach work https://lingui.js.org/tutorials/javascript.html#localizing-your-app ?
```js
i18n._(t`Network Nodes API request failed: ${response.code} ${response.message}:${responseMessage}`)
```

Reply:
So that we translate errors we throw `Network Nodes API request failed`

Reply:
Unfortunately it is not that easy. I created a follow up issue to establish a pattern to translate strings within dcos-ui https://jira.mesosphere.com/browse/DCOS-48301